src/Shared/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs

```diff
@@ -2,7 +2,11 @@
 {
     using ApplicationInsights.Extensibility;
     using Channel;
+#if AI_ASPNETCORE_WEB
     using Microsoft.ApplicationInsights.AspNetCore.Extensions;
+#else
+    using Microsoft.ApplicationInsights.WorkerService;
```

Comment:
is there any reason to ship it public in WorkService?

Reply:
we chose to make all initializers public quite some time back. It allows users to add/remove initializers as they need instead of being stuck with the default list.